### PR TITLE
Adjust audio and subtitle default selection

### PR DESCRIPTION
**Pull request by dezouzart**
### Goal
To trigger audioAvailable and subtitleAvailable events informing if has default value

### How to test
Run tests

### CODE REVIEW

File: Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 1]
```diff
@@ -450,16 +450,36 @@ open class AVFoundationPlayback: Playback {
...
         if let audioSources = self.audioSources {
-            trigger(.audioAvailable, userInfo: ["audios": audioSources])
+            let hasDefault = selectDefaultAudioIfNeeded()
+            trigger(.audioAvailable, userInfo: ["audios": audioSources, "hasDefaultFromOption": hasDefault])
         }
 
         addTimeElapsedCallback()
     }
 
+    fileprivate func selectDefaultSubtitleIfNeeded() -> Bool {
```
Comment (paulogama14) [inline Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Why do we need to return a Bool value?  I think that we have more than one responsibility in this method.

File: Tests/Clappr_Tests/AVFoundationPlaybackTests.swift [hunk 2]
```diff
@@ -820,6 +820,80 @@ class AVFoundationPlaybackTests: QuickSpec {
...
+                            let playback = AVFoundationPlayback(options: options)
+                            var hasDefaultFromOption = false
+                            playback.on(Event.subtitleAvailable.rawValue) { (userInfo: EventUserInfo) in
+                                guard let hasDefault = userInfo?["hasDefaultFromOption"] as? Bool else { return }
+                                hasDefaultFromOption = hasDefault
+                            }
+                            
+                            playback.play()
+                            
+                            expect(hasDefaultFromOption).toEventually(beTrue())
```
Comment (leandroalonso) [inline Tests/Clappr_Tests/AVFoundationPlaybackTests.swift]: This test is failing here. Increasing the timeout to `2` makes it pass.

File: Tests/Clappr_Tests/AVFoundationPlaybackTests.swift [hunk 3]
```diff
@@ -820,6 +820,80 @@ class AVFoundationPlaybackTests: QuickSpec {
...
+                            let playback = AVFoundationPlayback(options: options)
+                            var hasDefaultFromOption = true
+                            playback.on(Event.subtitleAvailable.rawValue) { (userInfo: EventUserInfo) in
+                                guard let hasDefault = userInfo?["hasDefaultFromOption"] as? Bool else { return }
+                                hasDefaultFromOption = hasDefault
+                            }
+                            
+                            playback.play()
+                            
+                            expect(hasDefaultFromOption).toEventually(beFalse())
```
Comment (leandroalonso) [inline Tests/Clappr_Tests/AVFoundationPlaybackTests.swift]: Same as before.

File: Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 4]
```diff
@@ -450,16 +450,36 @@ open class AVFoundationPlayback: Playback {
         seekOnReadyIfNeeded()
 
         if let subtitles = self.subtitles {
-            trigger(.subtitleAvailable, userInfo: ["subtitles": subtitles])
+            let hasDefault = selectDefaultSubtitleIfNeeded()
+            trigger(.subtitleAvailable, userInfo: ["subtitles": subtitles, "hasDefaultFromOption": hasDefault])
```
Comment (leandroalonso) [inline Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift]: `hasDefaultFromOption` is true if the user passed any audio/subtitle preference and it was selected, correct?

Maybe we can change this to `defaultSubtitleSelected`.

File: Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 5]
```diff
@@ -450,16 +450,36 @@ open class AVFoundationPlayback: Playback {
...
         }
 
         if let audioSources = self.audioSources {
-            trigger(.audioAvailable, userInfo: ["audios": audioSources])
+            let hasDefault = selectDefaultAudioIfNeeded()
+            trigger(.audioAvailable, userInfo: ["audios": audioSources, "hasDefaultFromOption": hasDefault])
```
Comment (leandroalonso) [inline Sources/Clappr_iOS/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Same as before. We can name `hasDefaultFromOption` to `defaultAudioSelected`.

File: Sources/Clappr/Classes/Base/AvailableMediaOptions.swift [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import AVFoundation
```
Comment (igoriols) [inline Sources/Clappr/Classes/Base/AvailableMediaOptions.swift]: No need to import `AVFoundation` here.

File: Sources/Clappr_tvOS/Classes/Plugin/Playback/AVFoundationPlayback/AVFoundationPlayback.swift [hunk 7]
```diff
@@ -455,6 +474,7 @@ open class AVFoundationPlayback: Playback {
         if keyPath == "currentItem.playbackLikelyToKeepUp" {
             if player?.currentItem?.isPlaybackLikelyToKeepUp == true && currentState == .buffering {
                 play()
+                selectDefaultSubtitleIfNeeded()
```
[the remark this answers is not in the document]
Comment (dezouzart) [reply Sources/Clappr_tvOS/Classes/Plugin/Playback/AVFoundationPlayback/AVFoundationPlayback.swift]: AVPlayer does not support 2 different media option set on earlier moment. You can check in `Select the Desired Media Option` session in [apple doc](https://developer.apple.com/documentation/avfoundation/media_assets_playback_and_editing/adding_subtitles_and_alternative_audio_tracks).
So I decide to set audio as soon as possible and subtitles later.

**Pull request merged**